code/game/gamemodes/umbrage/umbrage.dm

```diff
@@ -0,0 +1,184 @@
+	U << "<i>Look for the info button in the top left of your screen if you need help.</i>"
+	return 1
+
+/datum/game_mode/proc/greet_veil(mob/living/V)
+	if(!V)
+		return
+	V << "<span class='velvet big'><b>ukq wna ieja jks</b></span>" //"you are mine now"
+	V << "<b>Your mind goes numb. Your thoughts go blank. You feel utterly empty. \n\
+	A mind brushes against your own. You dream.\n\
+	Of a vast, empty Void in the deep of space.\n\
+	Something lies in the Void. Ancient. Unknowable. It watches you with hungry eyes. \n\
+	Eyes filled with stars.\n\
+	You feel a vast consciousness slowly consume your own and the veil falls away.\n\
+	Serve the umbrages above all else. Your former allegiances are now forfeit. Their goal is yours, and yours is theirs.</b>"
+	V << "<i>Use <b>.a</b> before your messages to speak over the Mindlink. This only works across your current z-level.</i>"
+	V << "<i>Ask for help from your masters or fellows if you're new to this role.</i>"
+	V << sound('sound/magic/become_veil.ogg', volume = 50)
+	flash_color(V, flash_color = "#21007F", flash_time = 100)
+
+/datum/game_mode/proc/equip_umbrage(mob/living/U)
+	var/datum/umbrage/S = new
+	S.linked_mind = U.mind
+	U.mind.umbrage_psionics = S
+	S.link_to_new_body(U)
+	S.give_ability("howto", 1)
+	S.give_ability("divulge", 1)
+	return
+
+#warn Umbrages need antag datums when LeoZ finishes his rework
+/datum/game_mode/proc/antag_umbrage(mob/living/U, silent)
+	if(!U.mind)
+		return
+	if(!silent)
+		greet_umbrage(U)
+	var/datum/mind/M = U.mind
+	umbrages += M
+	umbrages_and_veils += M
+	M.special_role = "Umbrage"
+	return 1
+
+/datum/game_mode/proc/antag_veil(mob/living/U, silent)
+	if(!can_become_veil(U))
+		return
+	if(!silent)
+		greet_veil(U)
+	var/datum/mind/M = U.mind
+	veils += M
+	umbrages_and_veils += M
+	M.special_role = "Veil"
+	return 1
+
+///////////
+// PROCS //
+///////////
+
+/proc/is_umbrage(datum/mind/M)
+	for(var/V in ticker.mode.umbrages)
+		var/datum/mind/T = V
+		if(T == M)
+			return 1
+	return
+
+/proc/is_veil(datum/mind/M)
+	for(var/V in ticker.mode.veils)
+		var/datum/mind/T = V
+		if(T == M)
+			return 1
+	return
+
+/proc/is_umbrage_or_veil(datum/mind/M)
+	return is_umbrage(M) || is_veil(M)
+
+/proc/can_become_veil(mob/living/M)
+	if(is_veil(M))
+		return
+	if(M.stat == DEAD || !M.mind || !M.client)
+		return
+	if(M.isloyal())
+		M.visible_message("<span class='warning'>[M] seems to resist an unseen force!</span>", \
+						"<b>Your mind goes numb. Your thoughts go blank. You feel utterly empty. \n\
+						A mind brushes against your own. You dream.\n\
+						Of a vast, empty Void in the deep of space.\n\
+						Something lies in the Void. Ancient. Unknowable. It watches you with hungry eyes. \n\
+						Eyes filled with stars.</b>\n\
+						<span class='boldwarning'>And not a single fuck was given.</span>")
+		return
+	return 1
+
+/proc/is_umbrage_progenitor(datum/mind/M)
```

Comment:
proc with no content, a stub.

Wiki pages with no/little content are stubs.
When things lose limbs they have a stub.
When you use a ticket, you're left with a ticket stub.